src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -86,7 +86,9 @@ public SentryEvent Process(SentryEvent @event)
         // if there's no exception with a stack trace, then get the current stack trace
         if (@event.Exception?.StackTrace is null)
         {
-            var stackTrace = SentryStackTraceFactoryAccessor().Create();
+            var stackTrace = @event.SentryExceptions?.FirstOrDefault() is { } sentryEx
```

Comment:
This means that there are now events with no exception but a SentryException?

Reply:
@bitsandfoxes yes. When we detect a blocking call, we capture that as an Event that has a SentryException but not System.Exception (since none was thrown). See [here](https://github.com/getsentry/sentry-dotnet/blob/45c67b95174c637687a2b038d4b14f5f1fabbce3/src/Sentry/Ben.BlockingDetector/BlockingMonitor.cs#L53-L80) for the code. 

@bruno-garcia not sure I follow - I'm not sure what the server requirements are.

Reply:
> capture that as an Event that has a SentryException but not System.Exception (since none was thrown)

That makes sense. Thanks